manifests/init.pp

```diff
@@ -1,34 +1,23 @@
 # Main file for puppet-ssh
 class ssh (
-  $server_options         = {},
-  $server_match_block     = {},
-  $client_options         = {},
-  $users_client_options   = {},
-  $version                = 'present',
-  $storeconfigs_enabled   = true,
-  $validate_sshd_file     = $::ssh::params::validate_sshd_file,
-  $use_augeas            = false,
-  $server_options_absent = [],
-  $client_options_absent = [],
-  $use_issue_net         = false,
+  Hash $server_options = {},
+  Hash $server_match_block = {},
+  Hash $client_options = {},
+  Hash $users_client_options = {},
+  String $version = 'present',
+  Boolean $storeconfigs_enabled = true,
+  Boolean $validate_sshd_file = $::ssh::params::validate_sshd_file,
+  Boolean $use_augeas = false,
+  Array $server_options_absent = [],
+  Array $client_options_absent = [],
+  Boolean $use_issue_net = false,
 ) inherits ssh::params {
 
-  validate_hash($server_options)
-  validate_hash($server_match_block)
-  validate_hash($client_options)
-  validate_hash($users_client_options)
-  validate_bool($storeconfigs_enabled)
-  validate_bool($validate_sshd_file)
-  validate_bool($use_augeas)
-  validate_array($server_options_absent)
-  validate_array($client_options_absent)
-  validate_bool($use_issue_net)
-
   # Merge hashes from multiple layer of hierarchy in hiera
-  $hiera_server_options = hiera_hash("${module_name}::server_options", undef)
-  $hiera_server_match_block = hiera_hash("${module_name}::server_match_block", undef)
-  $hiera_client_options = hiera_hash("${module_name}::client_options", undef)
-  $hiera_users_client_options = hiera_hash("${module_name}::users_client_options", undef)
+  $hiera_server_options = lookup("${module_name}::server_options", Optional[Hash], 'hash', undef)
```

Comment:
The lookup merge strategy (actually all lookup options) need to be configurable by the users. There are legitimate reasons why in a profile I might want to do a lookup with a deep merge instead of a hash merge. This is true for lines 18-20 as well.

Reply:
I fully agree with you. However, when working on this PR, I did everything to _not_ change the behavior of the module.

Reply:
I agree that this shouldn't be tackled in the PR.
The real question is why hiera_* lookups are even done, it would be better to just leave it to the automatic mechanism, then we could set the merge strategy in Hiera data: https://puppet.com/docs/puppet/5.0/hiera_merging.html#configuring-merge-behavior-in-hiera-data

Reply:
@kendall-moore You're right. Can you create a new issue for that please?